Top/csmodule.c

```diff
@@ -607,7 +607,12 @@ int32_t csoundLoadModules(CSOUND *csound)
 
   /* opcodedir GLOBAL override **experimental** */
   if (csound->opcodedir != NULL) {
-    dname = csound->opcodedir;
+    char *opcdir = cs_strdup(csound, csound->opcodedir);
```

Comment:
I'm wondering if this would have an unintended side effect where if you call csoundReset and try to reuse an existing csound instance it'd lose it's opcodedir value?

Reply:
the opcodedir value is not lost? 

The problem may be calling free() on it instead of csound->Free(), since it has been allocated with  cs_strdup(). 

First I want to see if this stops the reported crash.

Reply:
First time you create it's strdup in csoundCreate, then it's replaced with one with cs_strdup. If 
Then if you call csoundReset iI imagine it would csound->Free() that value. Now if you try to start CSOUND again after reset, won't it try to read an invalid pointer from csound->opcodedir?

Reply:
No, once it's created with cs_strdup(), it then becomes part of the Csound memory mechanism and if it's freed, then csound->opcodedir will be NULL and the code will not run because of the check. So there's no problem with this code at all (what I said about calling free on it was wrong). The Csound engine would lose the override, but that is what Csound 6 did anyway, since reset() would clear every member back to defaults (NULL, in this case).

However, it may make sense to keep the override under reset() since there is no separate function to set it now (we only do it in csoundCreate()). In this case, we just free() the string in csoundDestroy(). I've pushed a PR with this change (#2015).